### ISSUE TITLE
Several error

### ISSUE THREAD

**Issue by harleckinz** [go-vikunja/n8n-vikunja-nodes#1]
Hello!

I really like Vikunja and I was so happy that it finally got an n8n support, but unfortunately I ran into several problems:

1) api token only available in unstable images
2) n8n 1.15.2 showing the following error lines:

NodeApiError: UNKNOWN ERROR - check the detailed error for more information
TypeError: Cannot read properties of undefined (reading 'getCredentials')

3) the vikunja trigger node shows up in the list as Vikunja like the send node too.
4) the vikunja trigger node not works at all. 
5) none of the lists works, but if i enter the ID the function starts

I run Vikunja behind Nginx Proxy Manager, so i use internal docker address as API, may that why it's faulty. (http://vikunja-api:3456/api/v1)

A button for selecting all of the permission would be nice at API token page.

Anyway, great work! I'm so excited about the possibilities what it gonna unlock.

Regards

**Comment by kolaente** [comment]
Does it work if you use the same address as the one you're using when using the web frontend?

> the vikunja trigger node not works at all.

That requires webhooks, which are also available in unstable only at this point.

> the vikunja trigger node shows up in the list as Vikunja like the send node too.

I noticed that as well, don't know why it does that. If you have any ideas, please share.

> none of the lists works, but if i enter the ID the function starts

Can you explain that? I don't think I understood you correctly here.

**Comment by harleckinz** [comment]
Nope, but I triggered the webhook in browser and I got the following error:

'The requested webhook "GET faba251d-2f0c-49de-8a01-202582a7fbfe/webhook" is not registered.'

Any other webhook gets registered except this one.

"This webhook is not registered for GET requests. Did you mean to make a POST request?"

> Does it work if you use the same address as the one you're using when using the web frontend?

I'm not really into n8n custom nodes.

> I noticed that as well, don't know why it does that. If you have any ideas, please share.

So for example I make a request for Get Many Tasks. I've got two options (By ID, By List). It's works only with ID, because
the list shows Credential error.

![Screenshot 2023-11-24 at 13-23-45 n8n - ▶️ gufflard](https://github.com/go-vikunja/n8n-vikunja-nodes/assets/25708009/2b86b63d-3390-4c2a-8a1f-51556f06e61b)

> Can you explain that? I don't think I understood you correctly here.

**Comment by kolaente** [comment]
> Nope, but I triggered the webhook in browser and I got the following error:
> 
> 'The requested webhook "GET faba251d-2f0c-49de-8a01-202582a7fbfe/webhook" is not registered.'
> 
> Any other webhook gets registered except this one.

Sounds like a n8n error to me :thinking: 

> I'm not really into n8n custom nodes.

I mean the Vikunja URL.

> So for example I make a request for Get Many Tasks. I've got two options (By ID, By List). It's works only with ID, because
the list shows Credential error.

Gotcha. Sounds like you can't make any requests though. Which Vikunja version are you using? Does it work with [try](https://try.vikunja.io)? Which n8n version are you using?

**Comment by harleckinz** [comment]
Maybe. But seems like the n8n webhook is a GET one while the Vikunja uses a POST one and usually all the webhook is POST.

> Sounds like a n8n error to me 🤔

Yep I can't get the list, but I can manually add ID and works. Latest unstable Vikunja version and n8n 1.15.2. I never tried the main version. Unfortunately I use NPM as proxy so these problems could be unique and cannot be reproduced with the main one, anyway I will give a try later.

> Gotcha. Sounds like you can't make any requests though. Which Vikunja version are you using? Does it work with 
> [try](https://try.vikunja.io/)? Which n8n version are you using?

**Comment by kolaente** [comment]
I've just tried the webhook node with try and it seems to work just fine. Seems like the problem is somewhere in your installation :/

**Comment by harleckinz** [comment]
I've tested with multiple docker selfhosted services and none of my incoming webhook works so that's not an issue here anymore. 

I ran into some kinda Nginx error. Also painful that I cannot use proxied domain only my internal docker addresses. 

Whenever I finish my investigation I will return to fully test this awesome node.

**Comment by kolaente** [comment]
> I ran into some kinda Nginx error. 

Which nginx error?

> Also painful that I cannot use proxied domain only my internal docker addresses.

You should be able to use your proxied domain (that's what I do).

**Comment by harleckinz** [comment]
My n8n cannot get webhooks from my selfhosted services such as Baserow, Vikunja, etc. The outgoing traffic seems to be great, I can get data from these services. The basic Webhook node works well and same for the Google Services like Sheets Trigger.

Whenever I tried to use a proxied domain I've got "ETIMEOUT: my external ip:443", same for the HTTP Request except I can add proxy and can get result.

**Comment by GinoTonello** [comment]
Unfortunately I always get a 401 message with GET MANY Tasks, but I have no problems with GET TASK.

I have already tried to use the API of https://try.vikunja.io/, but I get the same error message.

Can anyone help me with this?

If further information is needed, please let me know.

<img width="992" alt="Screenshot 8" src="https://github.com/go-vikunja/n8n-vikunja-nodes/assets/53956195/2579ac46-da5c-4206-9dc6-163b810934b0">

**Comment by kolaente** [comment]
Does the api token you're using have the permission to get many tasks?

**Comment by GinoTonello** [comment]
Already selected everything
<img width="322" alt="Screenshot 9" src="https://github.com/go-vikunja/n8n-vikunja-nodes/assets/53956195/738afed2-2142-403f-911a-94852143fb18">

**Comment by kolaente** [comment]
@GinoTonello Anything in Vikunja's logs?

**Comment by ubeyou** [comment]
experienced the same thing. any update to the community nodes?

**Comment by kolaente** [comment]
Is this still a problem with the latest release or the last unstable build?

**Comment by ubeyou** [comment]
0.22.0 build. API token issue most likely as login jwt token able to work.

**Comment by kolaente** [comment]
So the problem remaining is basically https://github.com/go-vikunja/api/issues/105?

**Comment by ubeyou** [comment]
yes. basically api token not authorized to get project tasks and many more, even though it's enabled during token creation.

**Comment by kolaente** [comment]
> I've got an idea that a function would be nice which gets all the overdue and upcoming tasks. That's probably an easy upgrade due the overdue tasks already collected and coming via email and the upcoming task also sorted in a dedicated menu item.

That's a filter on the "get all tasks" endpoint, which means it should be possible with the current node already. The API request from the frontend is this one:

https://try.vikunja.io/api/v1/tasks/all?sort_by[]=due_date&sort_by[]=id&order_by[]=asc&order_by[]=desc&filter_by[]=done&filter_value[]=false&filter_comparator[]=equals&filter_concat=and&filter_include_nulls=false&page=1

And for upcoming:

https://try.vikunja.io/api/v1/tasks/all?sort_by[]=due_date&sort_by[]=id&order_by[]=asc&order_by[]=desc&filter_by[]=done&filter_by[]=due_date&filter_by[]=due_date&filter_value[]=false&filter_value[]=2024-02-27T22:30:04.459Z&filter_value[]=2024-02-20T22:30:04.459Z&filter_comparator[]=equals&filter_comparator[]=less&filter_comparator[]=greater&filter_concat=and&filter_include_nulls=false&page=1

**Comment by harleckinz** [comment]
Yep, that's already working via API request but it could be nice to introduce in the n8n node as well without limiting to a single project.

![Screenshot 2024-02-21 at 07-48-19 n8n - ▶️ vikunja értesítő](https://github.com/go-vikunja/n8n-vikunja-nodes/assets/25708009/0e4edc15-f936-46c1-9d32-d22e2b53db97)

**Comment by kolaente** [comment]
Once the new filter syntax is released, this will be added to the n8n node.

**Comment by Niskeletor** [comment]
Hello again! I'm experiencing a similar issue as @GinoTonello .

When I make "Get" and "Get Many" requests, there are usually no problems. Also, when I target directly by ID, it works fine. However, the problem arises when I try to get information through the "From List" selection.

I receive the following error:
`"Could not load list
please check your credentials"`

My API token has full permissions. I'm using Vikunja version 0.23 and n8n versions 1.41 and 1.42. Additionally, I have just updated to the latest version of the Vikunja community node.

Is there a solution for this? As mentioned, targeting directly by ID works without any issue.

![imagen](https://github.com/go-vikunja/n8n-vikunja-nodes/assets/97298802/606e63a4-758c-4a0f-a4b2-edbacd836370)

**Comment by kolaente** [comment]
@Niskeletor do you see any errors in Vikunja's API logs?

**Comment by Niskeletor** [comment]
Sorry for my late reply. As you can see, if I select "From List," I can't get the list of available projects to select. But if I choose the ID, the API can give me a response.

What I'm trying to say is that in n8n, with the Vikuja plugin, if I choose the "From List" option, it doesn't show any projects; the list appears empty. However, if I select by ID, it does return information.

![imagen](https://github.com/go-vikunja/n8n-vikunja-nodes/assets/97298802/da8cc784-8c28-4b62-8839-1ab4f215a3da)

![imagen](https://github.com/go-vikunja/n8n-vikunja-nodes/assets/97298802/31a37a10-dbff-4bc5-a756-dc8543256085)

![imagen](https://github.com/go-vikunja/n8n-vikunja-nodes/assets/97298802/62b47ee2-f703-4fd5-99fe-1c8ed665b1dc)

**Comment by kolaente** [comment]
@Niskeletor you need to select a project from the list if you set the drop-down to "from list".

**Comment by Niskeletor** [comment]
Yep, thats the point , when I try to select one of them returns an empy list, with the msg error below :
Could not load list (first screenshot)

**Comment by kolaente** [comment]
Does the Vikunja API token has permission to read all projects?

Do you see any errors in Vikunja's logs?

**Comment by kolaente** [comment]
Closing as inactive, please ping or open a new issue with relevant information if you still have this problem.

**Comment by deathwing696** [comment]
Hi, did you finally resolve this issue? 
I'm experiencing the same thing as user Niskeletor.
When I select a task by Id I have no problem, but when I try through the list it tells me to check my credentials. The API token has all the permissions, I have checked it and created several tokens.
Also, I don't know if it's related or not but the triggers for, for example, creating a new task, never fire, no matter how many tasks I create.

**Comment by deathwing696** [comment]
> Does it work when you use the same token to make an api request to [the tasks endpoint](https://try.vikunja.io/api/v1/docs#tag/task/paths/~1tasks~1all/get)?
>
Yes, if I make a request, via Postman, with that API key to the endpoint to get all the tasks (the one you have linked) it works without problems:
![imagen](https://github.com/user-attachments/assets/0058d679-b940-4e72-9f43-9c08f7d3a75a)
The problem is when it has to get the tasks through vikunja node it gives the following error:
![imagen](https://github.com/user-attachments/assets/a71d0dc7-0481-462f-a055-3bea532d4a1d)

> Are the triggers created as webhooks in Vikunja?
>
I have managed to solve the triggers issue (I don't know how but since last night I have managed to get them to work)
